### ISSUE TITLE
tier 2 swag: [implement/api] return appropriate coupon codes to user on login

### ISSUE THREAD

**Issue by amberleyromo** [gatsbyjs/api.gatsbyjs.org#15]
-  [ ] if a user has met the criteria for tier 1, return tier 1 coupon code.
-  [ ] if a user has met the criteria for tier 2, and redeemed tier 1 but not tier 2, return a tier 2 coupon code.
- [ ] if a user has met the criteria for tier 2, and redeemed neither tier 1 nor tier 2, return a tier 1 and a tier 2 coupon (two coupons)
- [ ] if a user has met the criteria for both tiers, and claimed both coupon codes

**Comment by jlengstorf** [comment]
We should switch over to using the Shopify GraphQL API, because it's a little easier to use than the REST API for complex queries.

For example, we can create a customer using:

```gql
mutation {
  customerCreate(input: {
    acceptsMarketing: true
    firstName: "Pat"
    email: "pat@example.org"
    tags: ["one", "two"]
  }) {
    customer {
      id
    }
  }
}
```

And we can query for that customer using their ID to get all orders (and discount codes) like so:

```gql
{
  customer(id: "gid://shopify/Customer/123456789098") {
    orders(first: 100) {
      edges {
        node {
          discountCode
        }
      }
    }
  }
}
```

With the resulting JSON looking like this:
```json
{
   "data":{
      "customer":{
         "orders":{
            "edges":[
               {
                  "node":{
                     "discountCode":"DISCOUNTCODE1"
                  }
               },
               {
                  "node":{
                     "discountCode":"DISCOUNTCODE2"
                  }
               },
               {
                  "node":{
                     "discountCode":"DISCOUNTCODE3"
                  }
               }
            ]
         }
      }
   },
   "extensions":{
      "cost":{
         "requestedQueryCost":103,
         "actualQueryCost":6,
         "throttleStatus":{
            "maximumAvailable":1000.0,
            "currentlyAvailable":994,
            "restoreRate":50.0
         }
      }
   }
}
```

**Comment by amberleyromo** [comment]
Closed in #17
